libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/RequireMapboxNavigation.kt

```diff
@@ -0,0 +1,137 @@
+@file:JvmName("RequireMapboxNavigation")
+
+package com.mapbox.navigation.core.lifecycle
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.navigation.core.MapboxNavigation
+import kotlin.properties.ReadOnlyProperty
+import kotlin.reflect.KProperty
+
+/**
+ * Extension function to make it simple to create the [RequireMapboxNavigationDelegate]. Below are
+ * a couple examples of how you may use the delegate.
+ *
+ * Default can be used when [MapboxNavigationApp] is setup elsewhere.
+ * ```
+ * val mapboxNavigation by requireMapboxNavigation()
+ * ```
+ *
+ * Initialize the [MapboxNavigationApp] when you are ready to use it
+ * ```
+ * val mapboxNavigation by requireMapboxNavigation {
+ *   MapboxNavigationApp.setup(..)
+ * }
+ * ```
+ *
+ * Register subscriptions and setup MapboxNavigationApp
+ * ```
+ * private val mapboxNavigation by requireMapboxNavigation(
+ *   onResumedObserver = object : MapboxNavigationObserver {
+ *     override fun onAttached(mapboxNavigation: MapboxNavigation) {
+ *       mapboxNavigation.registerLocationObserver(locationObserver)
+ *       mapboxNavigation.registerRoutesObserver(routesObserver)
+ *     }
+ *     override fun onDetached(mapboxNavigation: MapboxNavigation) {
+ *       mapboxNavigation.unregisterLocationObserver(locationObserver)
+ *       mapboxNavigation.unregisterRoutesObserver(routesObserver)
+ *     }
+ *   }
+ * ) {
+ *   MapboxNavigationApp.setup(
+ *       NavigationOptions.Builder(this)
+ *           .accessToken(accessToken)
+ *           .build()
+ *     )
+ * }
+ * ```
+ *
+ * @see [RequireMapboxNavigationDelegate] for more details.
+ */
+fun requireMapboxNavigation(
+    onCreatedObserver: MapboxNavigationObserver? = null,
+    onStartedObserver: MapboxNavigationObserver? = null,
+    onResumedObserver: MapboxNavigationObserver? = null,
+    onInitialize: (() -> Unit)? = null,
+): ReadOnlyProperty<LifecycleOwner, MapboxNavigation> = RequireMapboxNavigationDelegate(
+    onCreatedObserver = onCreatedObserver,
+    onStartedObserver = onStartedObserver,
+    onResumedObserver = onResumedObserver,
+    onInitialize = onInitialize
+)
+
+/**
+ * Attaches a [LifecycleOwner] to [MapboxNavigationApp] and provides access to [MapboxNavigation].
+ *
+ * You can choose to call [MapboxNavigationApp.setup] in the [onInitialize]. You can also setup in
+ * the onCreate calls, or any call that happens before this delegate is accessed. The delegate will
+ * crash if the app is not setup and an attached has been created.
```

Comment:
Should it be like this?
```suggestion
 * crash if the app is not setup or an attached lifecycle has not been created.
```

Reply:
You're right, I reworded a little more 👍